docs/gcloud-auth.rst

```diff
@@ -27,6 +27,10 @@ Overview
 
         $ export GOOGLE_APPLICATION_CREDENTIALS="/path/to/keyfile.json"
 
+    The default project ID will be the project that keyfile.json was genereated
+    from. This can be overridded by setting the ``GCLOUD_PROJECT`` environment
+    variable.
```

Comment:
If not in `gcloud-auth`, then I'm not really sure of the best place to put it. Also I should update the examples with `Client(project='target-project')` since it wouldn't be required anymore?